vocabularies/Org.OData.JSON.V1.xml

```diff
@@ -72,6 +72,53 @@
         </Annotation>
       </Term>
 
+      <Function Name="query">
+        <Annotation Term="Core.Description" String="Query stream values of media type `application/json`, returning a stream value of media type `application/json`" />
+        <Annotation Term="Core.LongDescription">
+          <String>Extracts a JSON value, such as an array, object, or a JSON scalar value (string, number, boolean, or `null`) from the `input` JSON value:
+- If `path` only consists of child member and single subscript operators, it returns the identified single node within `input`, or `null` if no node is identified. 
+- If `path` potentially identifies multiple nodes within `input` (by using recursive descendant, wildcard, array subset, or array filter expressions), it returns an array containing the identified nodes, or an empty array if no node is identified. 
+- If `input` is not a valid JSON value, the function returns `null`.
+- If `path` is `null`, not a valid [JSONPath expression](#Path), or does not match the structure of `input`, the function returns `null`. 
```

Comment:
According to the [JSONPath Processing Model](https://datatracker.ietf.org/doc/html/draft-ietf-jsonpath-base-01#section-3.2) the result of executing a valid query is always a list of nodes, even `$` produces a list with the input as its only item. Which makes sense, because even simple property access `$.foo` can produce a list with more than one element because JSON allows duplicate member names, only I-JSON forbids it.

We could follow this in two ways:
- the result of `JSON.query` is always a single JSON text that contains an array
- the result of `JSON.query` is always an OData collection of JSON texts, and the function can be suffixed with an index segment to retrieve only a single JSON text: `JSON.query(...)/0`

Reply:
The JSON function in Javascript does not support duplicate member names.
```
> JSON.parse('{"a":1,"a":2}')
{ a: 2 }
```
Having to dive into a JSON array or an OData collection even for the simplest queries is too cumbersome for me. I would at least suggest a convenience function `JSON.querySingle` that returns a single object (or null if that is not possible).